package/MDAnalysis/analysis/hbonds/wbridge_analysis.py

```diff
@@ -0,0 +1,813 @@
+        if not remove_duplicates:
+            logger.warning("Hidden feature remove_duplicates=False activated: you will probably get duplicate H-bonds.")
+
+        debug = kwargs.pop('debug', None)
+        if debug is not None and debug != self.debug:
+            self.debug = debug
+            logger.debug("Toggling debug to %r", self.debug)
+        if not self.debug:
+            logger.debug("WBridge analysis: For full step-by-step debugging output use debug=True")
+
+        self._timeseries = []
+        self.timesteps = []
+
+        logger.info("checking trajectory...")  # n_frames can take a while!
+        try:
+            frames = np.arange(self.u.trajectory.n_frames)[self.traj_slice]
+        except:
+            logger.error("Problem reading trajectory or trajectory slice incompatible.")
+            logger.exception()
+            raise
+        verbose = _set_verbose(verbose=kwargs.get('verbose', None),
+                               quiet=kwargs.get('quiet', None),
+                               default=True)
+        pm = ProgressMeter(len(frames),
+                           format="WBridge frame {current_step:5d}: {step:5d}/{numsteps} [{percentage:5.1f}%]\r",
+                           verbose=verbose)
+
+        logger.info("Starting analysis (frame index start=%d stop=%d, step=%d)",
+                    (self.traj_slice.start or 0),
+                    (self.traj_slice.stop or self.u.trajectory.n_frames), self.traj_slice.step or 1)
+
+        for progress, ts in enumerate(self.u.trajectory[self.traj_slice]):
+            # all bonds for this timestep
+
+            # dict of tuples (atom.index, atom.index) for quick check if
+            # we already have the bond (to avoid duplicates)
+
+            frame = ts.frame
+            timestep = ts.time
+            self.timesteps.append(timestep)
+            pm.echo(progress, current_step=frame)
+            self.logger_debug("Analyzing frame %(frame)d, timestep %(timestep)f ps", vars())
+            if self.update_selection1:
+                self._update_selection_1()
+            if self.update_selection2:
+                self._update_selection_2()
+            if self.update_water_selection:
+                self._update_water_selection()
+
+            s1_frame_results_dict = {}
+            if self.selection1_type in ('donor', 'both') and self._water_acceptors:
+                self.logger_debug("Selection 1 Donors <-> Water Acceptors")
+                ns_acceptors = AtomNeighborSearch(self._water_acceptors)
+                for i, donor_h_set in self._s1_donors_h.items():
+                    d = self._s1_donors[i]
+                    for h in donor_h_set:
+                        res = ns_acceptors.search(h, self.distance)
+                        for a in res:
+                            donor_atom = h if self.distance_type != 'heavy' else d
+                            dist = self.calc_eucl_distance(donor_atom, a)
```

Comment:
FYI: `self.calc_eucl_distance()` and the angle function are going to change, we need to rewrite any of the hbond stuff to use proper PBC-aware functions.

Reply:
@orbeckst Thank you for the reminder. The calc_eucl_distance() is method is inherited from hbondanalysis module so should change automatically if hbondanalysis changes.

Reply:
We need a simple distance function for two coordinates that can do PBC, like
```python
d = mda.lib.distances.calc_bonds([a1.position], [a2.position], box=ts.dimensions)[0]
```
(just not like this to remove stuff that slows down the function). It should be implemented in C/cython.

Once we have that, we can use it here.